coderd/autobuild/lifecycle_executor.go

```diff
@@ -179,7 +179,7 @@ func (e *Executor) runOnce(t time.Time) Stats {
 						Reason(reason)
 					log.Debug(e.ctx, "auto building workspace", slog.F("transition", nextTransition))
 					if nextTransition == database.WorkspaceTransitionStart &&
-						ws.AutomaticUpdates == database.AutomaticUpdatesAlways {
+						useActiveVersion(templateSchedule, ws) {
```

Comment:
add a test case to lifecycle_executor_test.go --- this is AGPL so all it can do is show that we ignore the template level setting.